tfjs-backend-webgpu/src/kernels/Concat_impl.ts

```diff
@@ -94,6 +94,26 @@ export function concatImpl(
     return outInfo;
   }
 
+  // There is a storage buffer limitation in compute stage, one for output so
+  // the maximum for input is limits.maxStorageBuffersPerShaderStage - 1
+  const maxInputNum = backend.device.limits.maxStorageBuffersPerShaderStage - 1;
+  if (inputs.length > maxInputNum) {
+    let reducedInputs = [];
+    for (let i = 0; i < inputs.length / maxInputNum; i++) {
```

Comment:
What if `inputs.length % maxInputNum !== 0`? And in the while body, `subArray.length !== 0` is always true?

Reply:
If `inputs.length % maxInputNum !== 0`,  means the length of last subArray of inputs is less than maxInputNum.
e.g. inputs.length is 100 and maxInputNum is 7, `inputs.length % maxInputNum === 2`, the last part is `inputs.slice(98, 105)` equals to `[inputs[98], inputs[99]]`
> And in the while body, subArray.length !== 0 is always true?

Yes, this check is unnecessary will remove it, Thanks.

Reply:
Number comparison in for loop condition used float type, so it will include the remaining part.
Thanks your advise to use `for (let i = 0; i < inputs.length; i += maxInputNum) {` instead `for (let i = 0; i < inputs.length/maxInputNum; i ++) {`, this reduced multiply and divide operation per loop.

Reply:
Good to know! LGTM, thanks!